limits.c

```diff
@@ -116,25 +116,18 @@ void limits_update_homing_values(uint8_t cycle_mask, float * homing_rate, float
 // Called from limits_go_home
 void limits_plan_homing(uint8_t cycle_mask, float homing_rate, float max_travel, uint8_t axislock, uint8_t approach, float* target, uint8_t flipped)
 {
- 
   uint8_t idx;
   
   // Set target location and rate for active axes.
   // and reset homing axis locks based on cycle mask.
 
   // limit travel distance to the length of the largest flag
   float travel = approach ? max_travel : MAXFLAGLEN;
+
   // set target for moving axes based on direction
   for (idx = 0; idx < N_AXIS; idx++) {
-    if (bit_istrue(cycle_mask, bit(idx))) {
-      if ((flipped & (1 << idx)) ^ approach) {
-        target[idx] = -travel;
-      } else {
-        target[idx] = travel;
-      }
-    } else {
-      target[idx] = 0;
-    }
+    const int8_t mult = ((flipped & (1 << idx)) ^ approach) ? -1 : 1;
+    target[idx] = bit_istrue(cycle_mask, bit(idx)) ? (mult * travel) : 0;
```

Comment:
I used to hate them as well, but they grew on me as a matter of keeping immutable state.  I think in this case it might not have been the best choice, but consider something a little more complex:

```
int i = 5;
string result = i % 2 == 0 ? "a"
              : i % 3 == 0 ? "b"
              : i % 5 == 0 ? "c"
              : i % 7 == 0 ? "d"
              : "e";
```

Which works great as a poor man's match operator.